src/main/java/com/unit3project/demo/Models/AbstractEntity.java

```diff
@@ -0,0 +1,52 @@
+package com.unit3project.demo.Models;
+
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Id;
+import jakarta.persistence.MappedSuperclass;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Size;
+import java.util.Objects;
+
+    @MappedSuperclass
+    public class AbstractEntity {
+
+        @Id
+        @GeneratedValue
+        private int id;
+
+        @Size(min=1, max = 100, message="Please enter a valid response between 1 and 100 characters.")
+        @NotBlank(message="Please enter a valid response between 1 and 100 characters.")
+        private String name;
+
+        @Override
+        public String toString() {
+            return name;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            AbstractEntity that = (AbstractEntity) o;
+            return getId() == that.getId();
+        }
+
+        @Override
+        public int hashCode() {
```

Comment:
line 36 for hashCode - should it be ..."id" and not "getId"?
@Override
    public int hashCode() {
        return Objects.hash(id)